src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2557,8 +2594,8 @@ internal bool EnableEventForDispatcher(EventDispatcher dispatcher, int eventId,
                 if (eventId >= m_eventData.Length)
                     return false;
 #if FEATURE_MANAGED_ETW
-                if (m_provider != null)
-                    m_eventData[eventId].EnabledForETW = value;
+                if (m_etwProvider != null)
+                    m_eventData[eventId].EnabledForBlobSerializedListeners = value;
```

Comment:
I don't see where we set EnabledForBlobSerializedListeners to true for the EventPipe case.

Reply:
Per our offline conversation, I have fixed this by having a boolean for each source (EventPipe and ETW) so that disabling one does not disable the other.

Reply:
I've also added a test to cover this case.